### PR TITLE
Add Points of Interest backend API: /otp/routers/default/pois

### LINKED ISSUE #204: Layer colors need to be updated
> - Bus stop D and bike lane info are the same colors (red).
> - Bus route E is similar in color to the streets and may cause confusion for colorblind users.

@jmfield2 is working on overhauling the color schemes, which should help with this.  Bus route E is probably the only question here, as that's the official Bull Runner color.  Related to #203, if we end up switching tile providers, as the street color could change.

### PR DESCRIPTION
**Pull request by jmfield2**
Filtering of returned points supported via "q" request parameter.

Map Builds now support build-config.json:

{
 pois: { amenity: {
                     parking: [],
                     bicycle_repair_station: [],
                     bicycle_parking: [],
                     charging_station: [],
                     fast_food: [],
                     restaurant: []
                  },
         shop:    {
                     supermarket: []
                  },
         sport: {
                    soccer: [],
                    baseball: [],
                    football: [],
                    softball: [],
                    basketball: [],
                    tennis: [],
                    running: []
                }
        }
}

And frontend UI configured using js/otp/config.js

layersWidget: {}
     { name: 'parking', target: '#parkinglots', icon: new L.layersIcon({icon: 'mdi mdi-parking', color: 'red'}), type: 'poi', search: 'amenity:parking', popupTemplate: 'parkingLots-template', condition: "permits", color: "red" },

```
Target is DOM selector defined in the client, typically in the layer widget template: usf-layer-menu in layers/layers-templates.html

Templates stored in js/otp/layers/layers-templates.html
Search matches key=value based on backend API "q" param
```

Route E color, and bike lanes updated fixes #204

Add links for EV stations for #227

Also for #177, #179

### CODE REVIEW

**Comment by jmfield2** [comment]
So, there are a handful of whitespace-related lines in this merge I think .. nothing major, but there.

Also, @jailby I think I regressed the 'box_col' class in layers-templates.html  @ https://github.com/CUTR-at-USF/usf-mobullity/pull/241/files#diff-bf2eecadf70c25f97650281de7bcfad1R125

File: src/client/js/otp/widgets/tripoptions/LayersWidget.js [hunk 1]
```diff
@@ -64,142 +64,166 @@ otp.widgets.LayersWidget =
         if (this.module.stopsLayer == undefined) $('.otplayerView-inner .stops').remove();
         if (this.module.busLayers == undefined) $('.otp-layerView-inner .bus').remove();
         if (this.module.bikeLayers == undefined) $('.otp-layerView-inner .bikes').remove();
-        
+
         // Layer group toggle code
-        
-	// Bullrunner bus stops+routes
+
+	    // Bullrunner bus stops+routes
```
Comment (jailby) [inline src/client/js/otp/widgets/tripoptions/LayersWidget.js]: This one

**Comment by jailby** [comment]
Yeah I saw the white space too.
I think it got read of the useless whitespace except one that I commented.

Yeah I see that you are not using `box_col` anymore. Can you briefly explain me why you did so? Was it the issue where the text needed to be on two lines?

**Comment by jmfield2** [comment]
No reason, I think I just used my changes over upstream during the merge
which is probably wrong .. I forgot if you changed that or not so I can
change them back to how they are on mobullity

**Comment by jailby** [comment]
hmm I do not remember changing that so I don't think it was me.

**Pull request merged**